cpp/src/arrow/ipc/adapter.cc

```diff
@@ -243,6 +271,26 @@ class RowBatchReader::Impl {
       return MakePrimitiveArray(
           type, field_meta.length, data, field_meta.null_count, null_bitmap, out);
     }
+    if (IsListType(type.get())) {
```

Comment:
Either else if, or I also think having more consistency in spacing will be better, so extra space after closing braces.